### PR TITLE
feat: add PinInput component

### PR DESCRIPTION
**Pull request by amje**

### CODE REVIEW

**Review by korvin89 — CHANGES_REQUESTED** [review]

File: src/components/PinInput/README.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+| otp              | When set to `true` adds `autocomplete="one-time-code"` to inputs                                                                                     |          `boolean`           |             |
+| placeholder      | Placeholder for inputs                                                                                                                               |           `string`           |             |
+| qa               | HTML `data-qa` attribute, for test purposes                                                                                                          |           `string`           |             |
+| size             | Size of input fields                                                                                                                                 |   `"s"` `"m"` `"l"` `"xl"`   |    `"m"`    |
+| style            | HTML `style` attribute                                                                                                                               |    `React.CSSProperties`     |             |
```
[the remark this answers is not in the document]
Comment (amje) [reply src/components/PinInput/README.md:183]: Root node is by default. Where prop is affecting input and not root node, I explicitly noted it

File: src/components/PinInput/PinInput.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    const refs = React.useRef<Record<number, HTMLInputElement | null>>({});
+    const [activeIndex, setActiveIndex] = React.useState(0);
+    const [focusedIndex, setFocusedIndex] = React.useState(-1);
+    const updateCallback = React.useCallback(
+        (newValue: string[]) => {
+            if (onUpdate) {
+                onUpdate(newValue);
+            }
+
+            if (onUpdateComplete && newValue.every((v) => Boolean(v))) {
```
[the remark this answers is not in the document]
Comment (amje) [reply src/components/PinInput/PinInput.tsx:76]: There can be only strings, `Boolean('0') === true`

File: src/components/PinInput/PinInput.tsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+        }
+        // We only care about autofocus on initial render
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    return (
+        <div ref={ref} className={b({size}, className)} style={style} data-qa={qa}>
+            <div className={b('items')}>
+                {Array.from({length}).map((__, i) => (
+                    <div key={i} className={b('item')}>
```
[the remark this answers is not in the document]
Comment (amje) [reply src/components/PinInput/PinInput.tsx:234]: Nope, TextInput itself has `width: 100%` style which break flex layout

File: src/components/PinInput/PinInput.tsx [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+                            controlProps={{
+                                inputMode: type === 'numeric' ? 'numeric' : 'text',
+                                pattern: type === 'numeric' ? '[0-9]*' : '[0-9a-zA-Z]*',
+                                className: b('control'),
+                                'aria-label': props['aria-label'],
+                                'aria-labelledby': props['aria-labelledby'],
+                                'aria-describedby': ariaDescribedBy,
+                                'aria-invalid': validationState === 'invalid' ? true : undefined,
+                            }}
+                            controlRef={handleRef.bind(null, i)}
```
[the remark this answers is not in the document]
Comment (amje) [reply src/components/PinInput/PinInput.tsx:256]: Yes. I only need curry the index

**Review by korvin89 — APPROVED** [review]

**Pull request merged**